### PR TITLE
[Freestanding] In the task-to-thread model, actors should behave like actual locks

### PR DESCRIPTION
**Pull request by rokhinip**
In the task-to-thread model in freestanding, have actors behave like actual locks.

Rdar: rdar://92348139

### CODE REVIEW

**Comment by rokhinip** [comment]
Oh whoops, I forgot to update the reflection stuff for distributed remote logic. Let me do that.

**Comment by rokhinip** [comment]
@swift-ci please test

**Review by mikeash — APPROVED** [review]

File: stdlib/public/Concurrency/Actor.cpp [hunk 1]
```diff
@@ -839,19 +833,25 @@ static_assert(sizeof(ActiveActorStatus) == ACTIVE_ACTOR_STATUS_SIZE,
...
+  bool isDistributedRemoteActor;
 
 public:
   /// Properly construct an actor, except for the heap header.
   void initialize(bool isDistributedRemote = false) {
-    ActiveActorStatus status = ActiveActorStatus();
-    if (isDistributedRemote) {
-      status = status.withDistributedRemote();
-    }
-    _status().store(status, std::memory_order_relaxed);
```
[the remark this answers is not in the document]
Comment (rokhinip) [reply stdlib/public/Concurrency/Actor.cpp:854]: Oh I might have mistakenly assumed that these objects are actually auto-initialized to 0. Is that not true in C++?

[the remark this answers is not in the document]
Comment (mikeash) [reply stdlib/public/Concurrency/Actor.cpp:854]: It is not!

**Comment by rokhinip** [comment]
@swift-ci please test

**Comment by rokhinip** [comment]
@swift-ci please test

**Comment by rokhinip** [comment]
@swift-ci please test

**Comment by rokhinip** [comment]
@swift-ci please test and merge

**Pull request merged**